src/main/java/edu/cornell/kfs/gl/businessobject/lookup/CuCurrentAccountBalanceLookupableHelperServiceImpl.java

```diff
@@ -0,0 +1,191 @@
+package edu.cornell.kfs.gl.businessobject.lookup;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+import org.kuali.kfs.coa.businessobject.Account;
+import org.kuali.kfs.gl.businessobject.Balance;
+import org.kuali.kfs.gl.businessobject.CurrentAccountBalance;
+import org.kuali.kfs.gl.businessobject.lookup.CurrentAccountBalanceLookupableHelperServiceImpl;
+import org.kuali.kfs.krad.util.ObjectUtils;
+import org.kuali.kfs.sys.KFSConstants;
+import org.kuali.kfs.sys.KFSParameterKeyConstants;
+import org.kuali.kfs.sys.businessobject.SystemOptions;
+import org.kuali.rice.core.api.util.type.KualiDecimal;
+
+import edu.cornell.kfs.sys.CUKFSParameterKeyConstants;
+
+/**
+ * Custom CurrentAccountBalance lookupable helper subclass that adds some enhancements
+ * to the lookup. One is an option for excluding "CB" amounts from certain calculations,
+ * effectively turning the results into Current Fund Balances instead. Another change
+ * is using object type code to perform Current Asset/Liability calculations instead,
+ * and there are associated parameters to go along with that. Some miscellaneous
+ * clean-up has been done as well.
+ */
+public class CuCurrentAccountBalanceLookupableHelperServiceImpl extends CurrentAccountBalanceLookupableHelperServiceImpl {
+
+    private static final long serialVersionUID = 2542719296293895780L;
+
+    /**
+     * This override copies the superclass's version of the method, and tweaks it to add the CB exclusion option
+     * and to use object type code for Current Asset/Liability checking.
+     * It also fixes a comparison bug in the Current Budget calculation that was getting the right value from the wrong constant.
+     * 
+     * @see org.kuali.kfs.gl.businessobject.lookup.CurrentAccountBalanceLookupableHelperServiceImpl#updateCurrentBalance(
+     * org.kuali.kfs.gl.businessobject.CurrentAccountBalance, org.kuali.kfs.gl.businessobject.Balance, java.lang.String)
+     */
+    @SuppressWarnings("deprecation")
+    @Override
+    protected void updateCurrentBalance(CurrentAccountBalance currentBalance, Balance balance, String fiscalPeriod) {

```

Comment:
This is a really long method, can we refactor this to make smaller functions that will be easier to ready